pageserver/src/tenant/timeline.rs

```diff
@@ -3641,7 +3636,7 @@ impl Timeline {
         let mut key_values_total_size = 0u64;
         let mut dup_start_lsn: Lsn = Lsn::INVALID; // start LSN of layer containing values of the single key
         let mut dup_end_lsn: Lsn = Lsn::INVALID; // end LSN of layer containing values of the single key
-        for (key, lsn, value_ref) in all_values_iter {
+        for &(key, lsn, _size, ref value_ref) in all_values_iter {
```

Comment:
Is the `ref` really needed...?

Reply:
Before this PR, we iterated over the entries by-value because we'd have to vecs. Now both iterators access the same vec so it has to happen by-ref. The key, lsn, size fields are all rather small and impl `Copy`, and current code inside the `for` assumes that key and lsn are not references. I can either add *'s in the for's body, or do this. I prefer latter as the code is nicer to read. The `value_ref` doesn't implement `Copy`, as it has an Arc internally, so its usage happens via references, which isn't much of a problem either, but yeah it adds the `&` and `ref`. But it's one time, instead of having to do it for most uses of key and lsn.

Reply:
No worries, I just failed to see it.